HISTORY.md

```diff
@@ -4,6 +4,7 @@
 * Fix wrong result being read from ingested file. May happen when a key in the file happen to be prefix of another key also in the file. The issue can further cause more data corruption. The issue exists with rocksdb >= 5.0.0 since DB::IngestExternalFile() was introduced.
 * Finish implementation of BlockBasedTableOptions::IndexType::kBinarySearchWithFirstKey. It's now ready for use. Significantly reduces read amplification in some setups, especially for iterator seeks.
 * Fix a bug by updating CURRENT file so that it points to the correct MANIFEST file after best-efforts recovery.
+* Fix a bug caused by trying to dereference nullable pointer (timestamp argument) in DB::MultiGet.
```

Comment:
I think this is both too scary and not scary enough.

It is too scary because the bug should only affect cases in which ReadOptions::timestamp is set, right? There's technically UB for all callers using default timestamps argument, but I don't think it's an issue in practice.

It's not scary enough because this wording suggests people are only affected if they've seen a crash. I mentioned how this could but could lead to wrong result. Here's the code: https://github.com/pdillinger/rocksdb/tree/fix-nullptr-multiget-alt

Reply:
Actually there are two bugs here. 1) trying to deref a nullptr, 2) incorrectly construction of LookupKey during MultiGet with timestamp. The wrong result you observed comes from 2).
Will update the HISTORY.md accordingly.

Reply:
I see on the wiki:

> Note that if the database enables timestamp, then caller of Get() should set ReadOptions.timestamp.

> When using Put with user timestamp, the user needs to set WriteOptions.timestamp.

My quick testing indicates these are not enforced with debug-build assertions. Why not?

Reply:
This PR fixes 3 places, thus updated HISTORY.md with 3 additional entries. Hopefully this is fine. Thanks @pdillinger for the review.